src/mongocxx/options/aggregate.hpp

```diff
@@ -108,6 +108,30 @@ class MONGOCXX_API aggregate {
     ///
     const stdx::optional<bsoncxx::document::view_or_value>& collation() const;
 
+    ///
+    /// Sets the variable mapping for this operation.
+    ///
+    /// @param let
+    ///   The new variable mapping.
+    ///
+    /// @return
+    ///   A reference to the object on which this member function is being called. This facilitates
+    ///   method chaining.
+    ///
+    /// @see https://docs.mongodb.com/manual/reference/command/aggregate/
+    ///
+    aggregate& let(bsoncxx::document::view_or_value let);
+
+    ///
+    /// Retrieves the current variable mapping for this operation.
+    ///
+    /// @return
+    ///   The current variable mapping.
+    ///
+    /// @see https://docs.mongodb.com/manual/reference/command/aggregate/
+    ///
+    const stdx::optional<bsoncxx::document::view_or_value>& let() const;
```

Comment:
Nice work putting together a tricky signature-- especially cool that you marked it const with respect to the object! 

Looking at the signature for these optional<> members, I assume that the intent is to represent a member that may or may not be "present", not to restrict someone from fiddling with the value. (The returned optional<> is const, but the value it contains is not. So modifying the contents will modify the data in the object. It looks like this is intended, so I think we're ok.)